31042 search box

bienvu opened the pull request:
Add search form style.

https://agov_pr491.agov.com.au/styleguide/agov_whitlam/section-components.html#kssref-components-search

joannguyen commented:
test this please

joannguyen commented:
test this please

bienvu commented:
Hi John,
I merged,fixed conflict and used $font-bold for pager.

JohnAlbin commented on themes/agov/agov_whitlam/sass/components/search/_search.scss:
```diff
@@ -0,0 +1,149 @@
+//
+// .search-form--light - Alternate search light styling.
+//
+// Style guide: components.search
```
Needs work?

bienvu commented:
Ok. i removed @TODO needs work.

JohnAlbin commented on themes/agov/agov_whitlam/sass/_variables.scss:
```diff
@@ -96,6 +96,8 @@ $menlo:           Menlo, 'DejaVu Sans Mono', 'Ubuntu Mono', Courier, 'Courier Ne
 // Now create some variables using the font stacks we want on this site.
 $font-body:       $opensans;
+$font-heading:    $opensansbold;
+$font-monospace:  $menlo;
```
You don't need to add these variables. The existing $font-* variables will work for whatever you need.

JohnAlbin commented on themes/agov/agov_whitlam/sass/_variables.scss:
```diff
@@ -122,6 +124,9 @@ $chroma: add-colors((
   black:                  #000,
+  grey-bb:                #bbb,
+  greyee:                 #eee,
+  greyfa:                 #fafafb,
```
I've noticed there's some slight color variations in the PSD that shouldn't be there. That's causing us to have colors that are imperceptibly different from an existing color.

So, instead of creating `grey-bb` use `grey-light`.
Instead of `greyee`, use `grey-extra-light`.
Instead of `greyfa`, use `white-smoke`.

joannguyen replied:
Hi John, Do you mean we can change the color on the existing color variations?
should we do like this:
   grey-dark:              #bbb,
   grey-extra-dark:     #eee,
   white-smoke:          #fafafb,

JohnAlbin replied:
No, the current color values are already being used on other parts, so we should keep the color values as-is. The new colors are imperceptibly different than the old ones, so there's no need for new color values.

JohnAlbin commented on themes/agov/agov_whitlam/sass/components/pager/_pager.scss:
```diff
@@ -46,6 +46,7 @@
     margin: 0 5px 0 0;
     border: 1px solid color(border);
     padding: 6px 12px;
+    font-family: $font-bold;
```
Why is this in this PR? Revert please.

JohnAlbin commented on themes/agov/agov_whitlam/sass/components/search/_search.scss:
```diff
@@ -0,0 +1,149 @@
+      &:-ms-input-placeholder {  
+        color: color(search-color); 
+			}
+  	}
```
Can we fix this indentation?

JohnAlbin commented on themes/agov/agov_whitlam/sass/components/search/search.hbs:
```diff
@@ -0,0 +1,16 @@
+      </div>
+    </div>
+  </form>
+</div>
```
End of file line ending, please.

JohnAlbin commented on themes/agov/agov_whitlam/sass/components/search/_search.scss:
```diff
@@ -0,0 +1,149 @@
+}
+
+.search-form--light,
+%search-form--light {
```
Why are we using a separate `.search-form--light` class? Shouldn't this just be the styling of the search-form on larger screens (i.e. a media query) rather than a different class?

joannguyen replied:
Hey John, the builders can use class .search-form--light if they put search block on dark background.
and we can extend class .search-form--light for mobile in this case.
Does it make sense?

JohnAlbin replied:
> and we can extend class .search-form--light for mobile in this case.

Extend it how? `@extend`? because `@extend` doesn't work with media queries.

joannguyen replied:
ok so we can not extend in media queries.
Bien did mobile first for search style so Bien need to fix style for large screen now.
Should we remove style for .search-form--light class or not? I thought it is good for the users later, not only for special case in design now

bienvu commented:
- Removed var color and revered pager like 7.x-3.x and fixed space 
- Search box style: I still keep 2 styles for search box.
* (1) If the users use search box in sidebar, content (light background), they dont need to add class. 
* (2) If the users use search box in header (dark background), they need to add class .search-form--light. In mobile I fixed style back to normal style (1). it adapts to special case in design now.

Desktop-search:
![desktop-search](https://cloud.githubusercontent.com/assets/8131755/7879034/8fa81d70-0616-11e5-9be1-1e65e811103f.png)

Mobile-search:
![mobile-search](https://cloud.githubusercontent.com/assets/8131755/7879038/a23fd59a-0616-11e5-92f8-559d27194919.png)

Let me know your thought John. Thanks

joannguyen commented:
Hi John, I see we can not put class in search box block now because search box is in sitewide block regions. So the simple way is styling default search box, don't need 2 styles and put class any more. 
Bien fixed it. can you please check and merge. Thanks.

JohnAlbin commented:
GitHub ticket # is wrong. Says 31042, but should be 31041
